frontend/src/components/Results/Results.js

```diff
@@ -0,0 +1,61 @@
+import "./Results.scss";
+import React, { useEffect, useState, useContext } from "react";
+import { Button } from "@material-ui/core";
+import { useHistory, useLocation } from "react-router";
+import useWindowSize from "react-use/lib/useWindowSize";
+import Confetti from "react-confetti";
+import LoggedContext from "../../contexts/LoggedContext";
+
+const Result = () => {
+  const history = useHistory();
+  const isLogged = useContext(LoggedContext);
+  const location = useLocation();
+  const [place, setPlace] = useState();
+  const { width, height } = useWindowSize();
+
+  useEffect(() => {
+    if (!isLogged) {
+      history.replace("/login");
+    }
+    if (!location.state) {
+      history.replace("/menu");
+    } else {
+      setPlace(location.state.place);
+    }
+  }, [isLogged, history, location.state]);
```

Comment:
This can be risky, not saying this will not work but keep in mind that `useEffect` only shallow compares values. This means that if `history` and `location.state` are objects and all their keys and values stay the same but the reference changes, this effect will rerun. `{} === {}` in javascript equals `false`. Just keep this in mind in case you get weird bugs/effects.

Reply:
Ok, I will remember and try to change it later.

Reply:
I just added it because of react warnings.